R/render.r

```diff
@@ -94,18 +94,43 @@ data_template <- function(pkg = ".", depth = 0L) {
 
 data_open_graph <- function(pkg = ".") {
   pkg <- as_pkgdown(pkg)
-  og <- list()
-  logo <- find_logo(pkg$src_path)
-  if (!is.null(logo)) {
+  og <- pkg$meta$template$opengraph %||% list()
+  og <- keep_supported_open_graph(og)
+  if (is.null(og$image) && !is.null(find_logo(pkg$src_path))) {
+    og$image <- list(src = path_file(find_logo(pkg$src_path)))
+  }
+  if (!is.null(og$image) && !grepl("^http", og$image$src)) {
     site_url <- pkg$meta$url %||% "/"
     if (!grepl("/$", site_url)) {
       site_url <- paste0(site_url, "/")
     }
-    og$image <- paste0(site_url, path_file(logo))
+    og$image$src <- gsub("^man/figures/", "reference/figures/", og$image$src)
+    og$image$src <- paste0(site_url, og$image$src)
+  }
+  if (length(og$twitter) &&
```

Comment:
The `length()` test is throwing me off. What data type do you expect `og$twitter` to be?

Reply:
I agree this is awkward. I've restructured this code and moved it to `check_open_graph()`. I took the opportunity to try to improve the error messages.